spinach/brokers/redis_scripts/enqueue_jobs_from_dead_broker.lua

```diff
@@ -14,33 +14,29 @@ local jobs_json = redis.call('hvals', running_jobs_key)
 
 for _, job_json in ipairs(jobs_json) do
     local job = cjson.decode(job_json)
-    if job["retries"] < job["max_retries"] then
-        job["retries"] = job["retries"] + 1
```

Comment:
I think it might be better to unconditionally increment `retries` here, rather than treat it as a "mulligan".

As an unrealistic example from my fevered imagination: a job that happens to kill its worker somehow every time it runs would just endlessly re-queue itself and never really leave any indication that something has gone wrong.  If it's incrementing the `retries` counter, there would at least be some forensics to use when looking around.

Reply:
I figured this might be controversial. Blindly incrementing retries is what gets you into the problematic situation in the first place and if you're looking for forensics I'd say your log file will be full of them.

I suppose we do need a way to clean out bad jobs that keep killing the worker (cough cough) so if you have any ideas, please say now.

The only other thing that came to mind as an alternative is to make the failure handler run when the last retry is passed. That would require some extensive changes here I think which is why I went this way first. But the bottom line for me is that I didn't think we should couple broker death with job failure.

Reply:
Fair enough.